### ISSUE TITLE
captured data is not in pcap format.

### ISSUE THREAD

**Issue by abhijit-dhar** [open-traffic-generator/ixnetwork#55]
```
pcap_bytes = api.get_capture_results(CaptureRequest(port_name=rx_port.name))
 with open('%s.pcap' % rx_port.name, 'wb') as fid:
            fid.write(pcap_bytes)

        from scapy.all import PcapReader
        reader = PcapReader('%s.pcap' % rx_port.name)
        for item in reader:
            logger.info(tem.time)
            logger.info(item.show())

```

It gives the following error:

/usr/local/lib/python2.7/dist-packages/scapy/utils.py:572: in __init__
    RawPcapReader.__init__(self, filename)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <scapy.utils.PcapReader instance at 0x7ff0363ba7a0>, filename = 'Rx.pcap'

    def __init__(self, filename):
        self.filename = filename
        try:
            self.f = gzip.open(filename,"rb")
            magic = self.f.read(4)
        except IOError:
            self.f = open(filename,"rb")
            magic = self.f.read(4)
        if magic == "\xa1\xb2\xc3\xd4": #big endian
            self.endian = ">"
        elif  magic == "\xd4\xc3\xb2\xa1": #little endian
            self.endian = "<"
        else:
>           raise Scapy_Exception("Not a pcap capture file (bad magic)")
E           Scapy_Exception: Not a pcap capture file (bad magic)

filename   = 'Rx.pcap'
magic      = 'M<\xb2\xa1'
self       = <scapy.utils.PcapReader instance at 0x7ff0363ba7a0>

/usr/local/lib/python2.7/dist-packages/scapy/utils.py:496: Scapy_Exception

**Comment by abhijit-dhar** [comment]
After discussion in the Subhajit + team: 
I look into the code to me it appeared that the files generated/created is in .cap format not .pcap

```
def get_capture_results(self, request):
        """Gets capture file and returns it as a byte stream
        """
        capture = self._vport.find(Name=request.port_name).Capture
        capture.Stop('allTraffic')
        time.sleep(3)

        payload = { 'arg1': [self._vport.href] }
        url = '%s/vport/operations/getCaptureInfos' % self._ixnetwork.href
        response = self._request('POST', url, payload)
        file_name = response['result'][0]['arg6']
        file_id = response['result'][0]['arg1']

        url = '%s/vport/operations/saveCaptureInfo' % self._ixnetwork.href
        payload = { 'arg1': self._vport.href, 'arg2': file_id }
        response = self._request('POST', url, payload)

        path = '%s/capture' % self._ixnetwork.Globals.PersistencePath
        url = '%s/files?absolute=%s&filename=%s.cap' % (self._ixnetwork.href, path, file_name)
        pcap_file_bytes = self._request('GET', url)
        return pcap_file_bytes
 
```
1. I have tried the following manually 1) took two arbitrary port 2) start capture 3) stop capture. 4) Tried to open the capture the capture file got downloaded in .cap format.
2. I also found that Wireshark is not installed in the ixNetwork webserver that I am using. 
3. Please let me know if Wireshark is a mandatory to execute the capture operations using Tgen API.

**Comment by ajbalogh** [comment]
1) the cap file being returned is in pcap format - the test https://github.com/open-traffic-generator/ixnetwork/blob/master/tests/test_capture_pcap.py successfully passes
2) wireshark is mandatory

**Comment by ajbalogh** [comment]
more info - capture does not require wireshark - it is only used for viewing the pcap/cap file - you can test your capture settings using the capture feature in the webui - it appears you have an empty capture file

**Comment by ajbalogh** [comment]
![image](https://user-images.githubusercontent.com/30300566/94834689-90571980-03c5-11eb-8826-b817dbe49d9b.png)

**Comment by ajbalogh** [comment]
cap/pcap are equivalent - what happens when you use the web ui?

**Comment by ajbalogh** [comment]
also can you upload the file to this issue

**Comment by abhijit-dhar** [comment]
Adding the Rx.pcap file in .zip format
[Rx.pcap.zip](https://github.com/open-traffic-generator/ixnetwork/files/5313617/Rx.pcap.zip)

**Comment by abhijit-dhar** [comment]
Actual error is that the pcap format we are creating is not supported by python 2.7. "pcap" format supported by python2.7 is "big endian" and "little endian" only. Python 3.x supports "big endian", "little endian", "big endian nanosecond-precision" and "little endian nanosecond-precision". The first four bytes of the pcap format determines the format of the pcap encoding. First four bytes of our pcap format 'M<\xb2\xa1' matches "little endian nanosecond precision". Since this is not supported in python 2.7 we are getting exception. Please change the format to support big/little endian only to be compatible with python 2.7.

```
 fp = open('Rx.pcap', 'rb')
magic = fp.read(4)
print(magic)
if magic == b"\xa1\xb2\xc3\xd4":  # big endian
    endian = ">"
    nano = False
    print("# big endian")
elif magic == b"\xd4\xc3\xb2\xa1":  # little endian
    endian = "<"
    nano = False
    print("# little endian")
elif magic == b"\xa1\xb2\x3c\x4d":  # big endian, nanosecond-precision
    endian = ">"
    nano = True
    print("nanosecond-precision")
elif magic == b"\x4d\x3c\xb2\xa1":  # little endian, nanosecond-precision
    endian = "<"
    nano = True
    print("nanosecond-precision")
else:
    print("Exception")


```

**Comment by ajbalogh** [comment]
Actually it is supported in python "2.7" just not in the version that you are using.
I used 2.7.14 and it is supported.
What exact version of 2.7 is being used and is this the same version that is being used in the sonic env?

**Comment by abhijit-dhar** [comment]
I am using sonic environment only and it has the version Python 2.7.12

**Comment by ajbalogh** [comment]
what does pip list return for scapy - ie what version of scapy
to get a newer version of scapy you can do a pip install --upgrade scapy

**Comment by ajbalogh** [comment]
scapy 2.2.0.dev is a pre-release version from oct 2013 - the sonic mgmt docker should be upgraded to a released version 2.4.4 is preferable.
